source/Octopus.Tentacle.Client/TentacleClient.cs

```diff
@@ -33,6 +33,17 @@ public class TentacleClient : ITentacleClient
         {
             this.scriptObserverBackOffStrategy = scriptObserverBackOffStrategy;
 
+            halibutRuntime.OverrideErrorResponseMessageCaching += response =>
```

Comment:
What does `+=` do here?

Reply:
Chains delegate handlers but the result from the last handler wins

![image](https://github.com/OctopusDeploy/OctopusTentacle/assets/86938706/a2387bc8-00a6-40e1-a26d-3fadf451cf63)

Reply:
Why did you opt to chain it over just replacing what it has?

Reply:
It doesn't make that much sense in this scenario. Instead, it makes more sense to actually chain to allow the caller to hook up their own error handling if needed e.g. Server wanted to do something additional to error caching.

```
var innerHandler = halibutRuntime.OverrideErrorResponseMessageCaching;
            halibutRuntime.OverrideErrorResponseMessageCaching = response =>
            {
                if (BackwardsCompatibleCapabilitiesV2Helper.ExceptionTypeLooksLikeTheServiceWasNotFound(response.Error.HalibutErrorType) ||
                    BackwardsCompatibleCapabilitiesV2Helper.ExceptionMessageLooksLikeTheServiceWasNotFound(response.Error.Message))
                {
                    return true;
                }

                // TentacleClient doesn't own the HalibutRuntime so allow other handlers to be configured that override the error caching behaviour
                return innerHandler?.Invoke(response) ?? false;
            };
```

Reply:
I've changed this to make more sense